src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +280,839 @@ static DamageTable D_808792D4 = {
+            func_80878724(this);
+        } else if (this->actor.xzDistToPlayer > SQ(10.0f) * this->unk_334) {
+            func_80877494(this);
+        } else {
+            func_80876B08(this, globalCtx);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80876BD0.s")
+void func_8087864C(EnDodongo* this) {
+    SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_06001A44, -4.0f);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_J_DAMAGE);
+    this->unk_302 = 0;
+    this->unk_304 = 0;
+    this->actor.speedXZ = 0.0f;
+    func_800BCB70(&this->actor, 0x4000, 0xFF, 0, 8);
+    this->actionFunc = func_808786C8;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80876CAC.s")
+void func_808786C8(EnDodongo* this, GlobalContext* globalCtx) {
+    if (this->unk_304 == 1) {
+        func_80876DC4(this, globalCtx);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80876D28.s")
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_80876B08(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80876DC4.s")
+void func_80878724(EnDodongo* this) {
+    SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_060013C4, -8.0f);
+    this->unk_302 = 0;
+    this->unk_304 = 0;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_J_DEAD);
+    this->actor.flags &= ~1;
+    this->actor.speedXZ = 0.0f;
+    func_800BCB70(&this->actor, 0x4000, 0xFF, 0, 8);
+    this->actionFunc = func_808787B0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_8087721C.s")
+void func_808787B0(EnDodongo* this, GlobalContext* globalCtx) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877278.s")
+    if (this->skelAnime.animCurrentFrame < 35.0f) {
+        if (this->unk_304 != 0) {
+            func_80876DC4(this, globalCtx);
+        }
+    } else if (this->actor.colorFilterTimer == 0) {
+        func_800BCB70(&this->actor, 0x4000, 0x78, 0, 4);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_808773C4.s")
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        if (this->unk_302 == 0) {
+            EnBom* bomb = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.world.pos.x,
+                                              this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, 0);
+            if (bomb != NULL) {
+                bomb->timer = 0;
+            }
+            this->unk_302 = 8;
+        }
+    } else if (func_801378B8(&this->skelAnime, 52.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GERUDOFT_DOWN);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877424.s")
+    if (this->unk_302 != 0) {
+        this->unk_302--;
+        if (this->unk_302 == 0) {
+            if (!this->actor.params) {
```

Comment:
```suggestion
            if (this->actor.params == 0) {
```
I've probably missed a few more of these.